src/gui/clientgui.cpp

```diff
@@ -1233,9 +1233,17 @@ void ClientGui::onDriveRemoved(int driveDbId) {
 
         // Erase drive
         driveInfoMapIt = _driveInfoMap.erase(driveInfoMapIt);
+
+        // Handle the case of the current drive being erased
         if (_currentDriveDbId == driveDbId) {
-            // Select next drive if there is one
-            _currentDriveDbId = !_driveInfoMap.empty() ? driveInfoMapIt->first : 0;
+            // Select the next drive if there is one.
+            if (driveInfoMapIt != _driveInfoMap.cend()) {
```

Comment:
This the fix: if the erased drive was the last item of the drive info map, then `driveInfoMapIt` is the `end()` iterator and cannot be dereferenced.

Reply:
Tested on Windows, all good 👍